plugins/ImageActions/main.lua

```diff
@@ -39,11 +42,97 @@ function cb(mode)
+      local width = filter:width()
+      local height = filter:height()
+      local last = vertical and height or width
+      local isWhite = {}
+
+      for i = 0, last - 1 do
+        if vertical then
+          isWhite[i+1] = filter:extract_area(0, i, width, 1):avg() > 254
+        else
+          isWhite[i+1] = filter:extract_area(i, 0, 1, height):avg() > 254
+        end
+      end
+
+      if #isWhite == 0 then
+        word = vertical and "rows" or "columns"
+        print("There are no completely white " .. word .. " in this image. Skipping.") 
+        table.insert(imdata, {data = im["data"], x=x, y=y, maxHeight=maxHeight, maxWidth=maxWidth})
+        goto continue 
+      end
+
+      local whiteEnds = {}
+      local whiteLengths = {}
+      local firstAfter = false
+      local currentLength = 0
+      local n = 0
+      for i = 1, last do
+        if isWhite[i] then
+          if currentLength > 0 then
+            currentLength = currentLength + 1
+          else 
+            currentLength = 1
+            firstAfter = true
+          end
+        else 
+          if firstAfter then
+            whiteEnds[n] = i - 1
+            whiteLengths[n] = currentLength
+            n = n + 1
+            firstAfter = false
+            currentLength = 0
+          end
+        end
+      end
+      whiteEnds[0] = nil
+      whiteLengths[0] = nil
+
+      if #whiteEnds == 0 then
+        print("Could not find a split")
+        table.insert(imdata, {data = im["data"], x=x, y=y, maxHeight=maxHeight, maxWidth=maxWidth})
+        goto continue
+      end
+
+      local maxLength = 0
+      local index = 0
+      for i = 1, #whiteEnds do
+        if whiteLengths[i] > maxLength then
+          maxLength = whiteLengths[i]
+          index = whiteEnds[i]
+        end
+      end
+      if vertical then
+        local image1 = image:extract_area(0, 0, width, index - maxLength // 2)
+        local image2 = image:extract_area(0, index - maxLength // 2, width, height - index + maxLength // 2)
+        table.insert(imdata, {data = image1:write_to_buffer(".png"), x=x, y=y, maxWidth=maxWidth})
+        table.insert(imdata, {data = image2:write_to_buffer(".png"), x=x, y=y+index*maxHeight/height, maxWidth=maxWidth})
+      else 
+        local image1 = image:extract_area(0, 0, index - maxLength // 2, height)
+        local image2 = image:extract_area(index - maxLength // 2, 0, width - index + maxLength // 2, height)
+        table.insert(imdata, {data = image1:write_to_buffer(".png"), x=x, y=y, maxHeight=maxHeight})
+        table.insert(imdata, {data = image2:write_to_buffer(".png"), x=x+index*maxWidth/width, y=y, maxHeight=maxHeight})
+      end
+      ::continue::
     end
-    table.insert(imdata, {data = image:write_to_buffer(".png"), x=x, y=y, maxWidth=maxWidth, maxHeight=maxHeight})
   end
   app.uiAction({action = "ACTION_DELETE"})
   app.addImages({images=imdata, allowUndoRedoAction = "grouped"})
```

Comment:
I think grouping these two to one `undoRedoAction` is currently not possible, is it? Would be nice for this case but not that easy to implement (and also not wanted since the api would need to keep some state from api-call to api-call)

Reply:
I haven't examined that, but I was also wandering why only the second piece was in the selection.

Reply:
Only the second image is put into a selection (which comes from `ImageHandler.addImageToDocument` only taking a single image), but the two images are undone grouped when hitting CTRL+z. Redo segfaults somehow, but that is an issue unrelated to this PR.